### ISSUE TITLE
Rook client fails to get nodes

### ISSUE THREAD

**Issue by DarkBlaez** [rook/rook#635]
Using kubeadm installed kubernetes cluster v1.6.2 on Centos 7.3 with 1 master 2 minons

Rook installs just fine with the operator and cluster.

Installed rook client which went fine as well

Then entered the following:

Connect to the rook-client pod 
kubectl -n rook exec rook-client -it bash

Verify the rook client can talk to the cluster:
rook node ls

The "rook node ls" resulted in:

failed to get nodes: HTTP status code 500 for query node: ''

Please advise

Thanks
DB

**Comment by jbw976** [comment]
Hi @DarkBlaez, do you happen to have the rook API server logs?  Something like:
```
kubectl -n rook logs `kubectl -n rook get pod | grep -F 'rook-api' | awk '{print $1}'`
```

**Comment by DarkBlaez** [comment]
Output from: kubectl -n rook logs `kubectl -n rook get pod | grep -F 'rook-api' | awk '{print $1}'`

2017-05-02 20:10:58.292593 I | cephmon: parsing mon endpoints: mon0=192.168.41.9:6790,mon1=192.168.197.203:6790,mon2=192.168.41.10:6790
2017-05-02 20:10:58.293619 I | cephmon: connecting to ceph cluster rook with user admin
2017-05-02 20:10:58.378598 I | cephmon: successfully connected to cluster rook with user admin
2017-05-02 20:15:12.522122 I | api: fb4de1e9-b755-4278-96cb-6a019bcb0e09	GET	/node	GetNodes	(start)
2017-05-02 20:15:12.522276 I | api-k8s: Getting nodes
2017-05-02 20:15:12.572130 E | api: failed to list nodes: failed to get nodes. User "system:serviceaccount:rook:default" cannot list nodes at the cluster scope. (get nodes)
2017-05-02 20:15:12.572192 I | api: fb4de1e9-b755-4278-96cb-6a019bcb0e09	GET	/node	GetNodes	500 Internal Server Error	51.055578ms
2017-05-02 20:15:19.356145 I | api: f8e8d109-1f48-4be9-aa28-8a17a4106d19	GET	/node	GetNodes	(start)
2017-05-02 20:15:19.356244 I | api-k8s: Getting nodes
2017-05-02 20:15:19.360040 E | api: failed to list nodes: failed to get nodes. User "system:serviceaccount:rook:default" cannot list nodes at the cluster scope. (get nodes)
2017-05-02 20:15:19.360115 I | api: f8e8d109-1f48-4be9-aa28-8a17a4106d19	GET	/node	GetNodes	500 Internal Server Error	4.024629ms
2017-05-02 20:18:09.428212 I | api: cc231926-c2c8-4a8d-8b78-a6b2cff15bd1	GET	/node	GetNodes	(start)
2017-05-02 20:18:09.428329 I | api-k8s: Getting nodes
2017-05-02 20:18:09.431934 E | api: failed to list nodes: failed to get nodes. User "system:serviceaccount:rook:default" cannot list nodes at the cluster scope. (get nodes)
2017-05-02 20:18:09.431980 I | api: cc231926-c2c8-4a8d-8b78-a6b2cff15bd1	GET	/node	GetNodes	500 Internal Server Error	3.843069ms
2017-05-02 20:19:30.894950 I | api: 39706ce6-3d0e-4043-b442-ff18a5fc1cf5	GET	/node	GetNodes	(start)
2017-05-02 20:19:30.895103 I | api-k8s: Getting nodes
2017-05-02 20:19:30.899067 E | api: failed to list nodes: failed to get nodes. User "system:serviceaccount:rook:default" cannot list nodes at the cluster scope. (get nodes)
2017-05-02 20:19:30.899350 I | api: 39706ce6-3d0e-4043-b442-ff18a5fc1cf5	GET	/node	GetNodes	500 Internal Server Error	4.505619ms
2017-05-02 20:31:20.177900 I | api: d4197c8d-d48f-4dba-ad13-b73b0f64a14b	GET	/pool	GetPools	(start)
2017-05-02 20:31:20.178887 I | cephmon: connecting to ceph cluster rook with user admin
2017-05-02 20:31:20.212187 I | cephmon: successfully connected to cluster rook with user admin
2017-05-02 20:31:20.224179 I | api: d4197c8d-d48f-4dba-ad13-b73b0f64a14b	GET	/pool	GetPools	200 OK	46.550762ms
2017-05-02 20:31:20.226842 I | api: 0f017490-b1b4-4c0a-9160-825f94c71292	POST	/pool	CreatePool	(start)
2017-05-02 20:31:20.227400 I | cephmon: connecting to ceph cluster rook with user admin
2017-05-02 20:31:20.244395 I | cephmon: successfully connected to cluster rook with user admin
2017-05-02 20:31:21.965456 I | cephclient: creating pool replicapool succeeded, info: pool 'replicapool' created, buf: 
2017-05-02 20:31:21.970530 I | api: 0f017490-b1b4-4c0a-9160-825f94c71292	POST	/pool	CreatePool	200 OK	1.743679986s

**Comment by jbw976** [comment]
Looks like a Kubernetes RBAC related issue where the API server pod doesn't have the permissions to do the node listing.  I can confirm that I'm seeing this too.

```
2017-05-02 20:15:12.572130 E | api: failed to list nodes: failed to get nodes. User "system:serviceaccount:rook:default" cannot list nodes at the cluster scope. (get nodes)
```
https://kubernetes.io/docs/admin/authorization/rbac/

**Comment by DarkBlaez** [comment]
Thanks. Will this have any impact on any rook functionality other than operating the client pod?

**Comment by jbw976** [comment]
I *think* that just `rook node ls` output is affected by this, since I *think* that's the only time we are trying to hit the `get nodes` API from Kubernetes.  Since enabling RBAC by default, we've given permissions for most everything else that I know of, with the exception of this and #612.  We'll try to get those sorted out.  Thanks again for reporting this issue!!

**Comment by DarkBlaez** [comment]
Thanks for the response

**Comment by kokhang** [comment]
@DarkBlaez In which namespace did you deploy the operator? If you use the rook-operator.yaml as is, then you have to deploy it in `default` namespace.

**Comment by DarkBlaez** [comment]
I did use the default rook-operator.yaml as is, no changes

**Comment by travisn** [comment]
This is related to #612 and will be fixed at the same time.

**Comment by DarkBlaez** [comment]
Given this, can I assume this is non-functional to use for storage use until milestone v0.5?

**Comment by travisn** [comment]
this change will actually be included in v0.4, which should be today

**Comment by DarkBlaez** [comment]
Great, once change is in I will spin up new VMs and test

**Comment by travisn** [comment]
fixed by #639. The `master-latest` tag now has the fix
